.github/workflows/ci.yml

```
@@ -0,0 +1,119 @@
+        fi
+
+    - name: Run tests (if source exists)
+      run: |
+        if [ -d "src/finbricklab" ] && [ "$(ls -A src/finbricklab)" ]; then
+          poetry run pytest --cov=src/finbricklab --cov-report=xml
+        else
+          echo "No source code found, skipping tests"
+        fi
+
+    - name: Upload coverage to Codecov
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)